### ISSUE TITLE
JVM doesn't exit after Cloud  is shutdown (except if running within a JUnit test)

### ISSUE THREAD

**Issue by hindog** [gridkit/nanocloud#20]
I can't seem to get the JVM to terminate after my `main` method exists event though I'm calling `cloud.shutdown()` just like the tutorial.  What's peculiar, though, is that I can't reproduce the issue when running within JUnit.  I've stepped through the code quite a bit with a debugger, but I'm having trouble tracking down why this is.

I created an example project with some test classes and documentation to reproduce what I'm experiencing here:

https://github.com/hindog/nanocloud-example

(Before running the examples, you'll want to tweak the `ExampleGrid` class to work with a remote node in your environment).  

By the way, I **am** able to work-around this by calling the following before `cloud.shutdown()`:
```
((HostControlConsole)cloud.node(remoteHost))
  .getPragma(ViConf.SPI_CONTROL_CONSOLE)).terminate();
```
... so it's not a "blocker" for me, it just seemed a pretty back hack on my part which makes me think I might be doing something wrong.

**Comment by aragozin** [comment]
Thanks for providing example. Not long after starting investigation I've stumbled on "`TODO Shutdown hook`" in RemoteConsoleInitializer.
I will publish fixed version on next week end.

**Comment by aragozin** [comment]
Fix is available in 0.8.15

**Comment by hindog** [comment]
Confirmed the fix on my end... thanks for the quick response!
